### ISSUE TITLE
maxNumberOutputs break ipdb

### ISSUE THREAD

**Issue by mlucool** [jupyterlab/jupyterlab#12321]
## Description

Ipdb output appears to count as mime renders. It removes the latest renders _and_ clicking to show complete output makes things worse by breaking future output.

## Reproduce

Set `maxNumberOutputs` to 50 and run this example:
```
import ipdb
def inf_loop():
    i = 0
    while True:
        print(i)
        i += 1
ipdb.runcall(inf_loop)
```
Press "n", then hold down the <ENTER> key until at least 50 "next" commands have been processed. If one clicks on the "click this message to get the complete output", it'll often show some of the buffered output. Press "n" again and hold <ENTER>. You'll see the scrollbar and output display fluttering around, but it just keeps deleting and reappending the output that appeared when clicking on the warning. No new output appears.

![16260-1-0_trimmed_ipdb_output](https://user-images.githubusercontent.com/1813603/161160819-8f0d0d07-c76e-4093-8746-e0c8f275439a.gif)

## Expected behavior

It works the same as `maxNumberOutputs=0`.

## Context
- JupyterLab version: 3.3.0

**Comment by telamonian** [comment]
Investigating this now. This is part of a larger issue with the functionality of the `maxNumberOutputs` option, which currently breaks any kind of `ipdb`-style "interactive" output

**Comment by telamonian** [comment]
There's now a description of the general issue at #12497, and a proposed fix at #12498
